lib/Backends/DeviceManagers.cpp

```diff
@@ -29,46 +29,52 @@ namespace runtime {
 /// when you define a new DeviceManager.
 
 /// Create a new instance of the interpreter Device.
-DeviceManager *createInterpreterDeviceManager(llvm::StringRef name);
+DeviceManager *
+createInterpreterDeviceManager(std::unique_ptr<DeviceConfig> config = nullptr);
```

Comment:
Why did you remove the name field? For some vendors, that might be used to differentiate between different devices (e.g. by PCIe bus name).

Reply:
That would go into the derived config object for that vendor. Since it's not global to all types I removed it.